### PR TITLE
Adding new features to primary branch

### PR DESCRIPTION
**Pull request by vodriemf**
US-292015 Providing Reason for Cancelled/Paused Scan on Polling for Results with Jenkins Plugin
US-294012 Providing Origin Source Field Data to API
US-316003 Updating Jenkins Marketplace Info and Developer Info
US-314001 Improving Jenkins Payload Packaging By Adding GUI Option
D-324012 Updating Incorrect Verbiage On Payload Location
US-318012 Adding Jenkins Scanning Priority Options In UI
D-324009 Eliminating Jenkins plugin returning NULLs instead of Cancel reason and Cancel reason notes
D-323004 Changing Logging in Reconnect Logic
US-313006 Removing UnNeeded Jenkins Plugin Options

### CODE REVIEW

**Comment by mtgibbs** [comment]
Can you please provide the features that you're adding to the description of this MR?  It's going to make it easier for you in the future when you're making release notes.

**Comment by mtgibbs** [comment]
Please be sure to format all of your files before this is taken.  There's inconsistency in style across each change. @vodriemf

**Comment by vodriemf** [comment]
@mtgibbs Completed both! Thanks for the input!

File: src/main/resources/org/jenkinsci/plugins/fodupload/steps/FortifyStaticAssessment/config.jelly [hunk 1]
```diff
@@ -31,34 +36,19 @@
...
+                <option value="RemediationScanOnly">Remediation Scan Only</option>
+                <option value="NonRemediationScanOnly">Non-Remediation Scan Only</option>
+            </select>
         </f:entry>
-        <f:entry title="Prefer Remediation if Available" field="isRemediationPreferred" help="/plugin/fortify-on-demand-uploader/help-isRemediationPreferred.html">
-            <f:checkbox/>
+        <f:entry title="Action when scan is in progress" field="inProgressScanActionType">
+            <select name="inProgressScanActionType">
+                <option value="DoNotStartScan">Do Not Start Scan</option>
+                <option value="CancelInProgressScan">Cancel Scan Then Start</option>
```
[the remark this answers is not in the document]
Comment (vodriemf) [reply src/main/resources/org/jenkinsci/plugins/fodupload/steps/FortifyStaticAssessment/config.jelly]: @mtgibbs  Makes sense. The FodEnums.java file is starting to get big. I'll make that the name of an enum folder and refactor the enums into individual files. The same advice can be given for the remediation scan options

**Pull request merged**